Pinta.Core/Managers/ToolManager.cs

```diff
@@ -58,71 +73,68 @@ public interface IToolService
 		/// 'PencilTool'. Returns a value indicating if tool was successfully changed.
 		/// </summary>
 		bool SetCurrentTool (string tool);
+
+		/// <summary>
+		/// Sets the current tool to the next tool with the specified shortcut.
+		/// </summary>
+		void SetCurrentTool (Gdk.Key shortcut);
 	}
 
 	public class ToolManager : IEnumerable<BaseTool>, IToolService
 	{
-		int index = -1;
-		int prev_index = -1;
-		
-		private List<BaseTool> Tools;
-		private Dictionary<Gdk.Key, List<BaseTool>> groupedTools;
-		private Gdk.Key LastUsedKey;
-		private int PressedShortcutCounter;
+		private readonly SortedSet<BaseTool> tools = new SortedSet<BaseTool> (new ToolSorter ());
```

Comment:
Minor style suggestion - I think with C# 9 this could be `private readonly SortedSet<BaseTool> tools = new (new ToolSorter ());`